src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -162,20 +162,18 @@ const ViewerStateProvider: React.FC<{ viewerSettings?: Partial<ViewerState> }> =
   // Sync viewer settings prop with state
   // React docs seem to be fine with syncing state with props directly in the render function:
   // https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
-  if (props.viewerSettings) {
-    let newSettings = viewerSettings;
-
-    for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
-      if (viewerSettings[key] !== props.viewerSettings[key]) {
-        // Update viewer settings one at a time to allow change handlers to keep state valid
-        newSettings = applyChangeToViewerSettings(viewerSettings, key, props.viewerSettings[key] as any);
+  useMemo(() => {
+    let newSettings = { ...viewerSettings };
+    if (props.viewerSettings) {
+      for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
+        if (newSettings[key] !== props.viewerSettings[key]) {
+          // Update viewer settings one at a time to allow change handlers to keep state valid
+          newSettings = applyChangeToViewerSettings(newSettings, key, props.viewerSettings[key] as any);
+        }
       }
     }
-
-    if (newSettings !== viewerSettings) {
-      setViewerSettings(newSettings);
-    }
-  }
+    setViewerSettings(newSettings);
+  }, [props.viewerSettings]);
```

Comment:
Converted to useMemo so this only runs when the viewer settings change. (Otherwise, this caused an infinite render loop.)